src/linear_algebra/basis.lean

```diff
@@ -888,6 +888,32 @@ show finsupp.total _ _ _ v _ = v i, by simp
 @[simp] lemma coe_mk : ⇑(basis.mk hli hsp) = v :=
 funext (mk_apply _ _)
 
+variables {hli hsp}
+
+/-- Given a basis, the `i`th element of the dual basis evaluates to 1 on the `i`th element of the
+basis. -/
+lemma mk_coord_apply_eq (i : ι) :
+  (basis.mk hli hsp).coord i (v i) = 1 :=
+show hli.repr ⟨v i, submodule.subset_span (mem_range_self i)⟩ i = 1,
+by simp [hli.repr_eq_single i]
+
+/-- Given a basis, the `i`th element of the dual basis evaluates to 0 on the `j`th element of the
+basis if `j ≠ i`. -/
+lemma mk_coord_apply_neq {i j : ι} (h : j ≠ i) :
+  (basis.mk hli hsp).coord i (v j) = 0 :=
+show hli.repr ⟨v j, submodule.subset_span (mem_range_self j)⟩ i = 0,
+by simp [hli.repr_eq_single j, h]
+
+/-- Given a basis, the `i`th element of the dual basis evaluates to the Kronecker delta on the
+`j`th element of the basis. -/
+lemma mk_coord_apply {i j : ι} :
+  (basis.mk hli hsp).coord i (v j) = if j = i then 1 else 0 :=
```

Comment:
You're right that there exists a more direct proof, e.g.,
```lean
lemma mk_coord_apply {i j : ι} :
  (basis.mk hli hsp).coord i (v j) = pi.single j 1 i :=
begin
  simp only [coord_apply, mk_repr, ← finsupp.single_eq_pi_single],
  erw hli.repr_eq_single j ⟨v j, submodule.subset_span (mem_range_self j)⟩ rfl,
end
```
but I'm not sure I prefer this and I still think it's worth having the special case `mk_coord_apply_eq` and `mk_coord_apply_ne` (albeit they would become corollaries rather than supporting lemmas).

I see `pi.single` here as baggage but I don't feel strongly so I'll happily make the change if you still lean in favour.